core/src/main/java/io/parsingdata/metal/Shorthand.java

```diff
@@ -164,10 +164,10 @@ private Shorthand() {}
     public static BinaryValueExpression mul(final ValueExpression left, final ValueExpression right) { return new Mul(left, right); }
     public static BinaryValueExpression sub(final ValueExpression left, final ValueExpression right) { return new Sub(left, right); }
     public static BinaryValueExpression mod(final ValueExpression left, final ValueExpression right) { return new Mod(left, right); }
-    public static UnaryValueExpression neg(final ValueExpression operand) { return new Neg(operand); }
+    public static OneToManyValueExpression neg(final ValueExpression operand) { return new Neg(operand); }
     public static BinaryValueExpression and(final ValueExpression left, final ValueExpression right) { return new io.parsingdata.metal.expression.value.bitwise.And(left, right); }
     public static BinaryValueExpression or(final ValueExpression left, final ValueExpression right) { return new io.parsingdata.metal.expression.value.bitwise.Or(left, right); }
-    public static UnaryValueExpression not(final ValueExpression operand) { return new io.parsingdata.metal.expression.value.bitwise.Not(operand); }
+    public static OneToManyValueExpression not(final ValueExpression operand) { return new io.parsingdata.metal.expression.value.bitwise.Not(operand); }
```

Comment:
Referring to your naming debate, reading this does make it seem a bit funny in that it now reads like `negate/not` have a different cardinality between input and output (consider an identity function, it would be even more weird imo), although both are simply (and mostly unbounded) ValueExpressions. Before, `Unary` meant a single operand, with implicitly a single output value (as always). Now I think it is a mix of two concepts. 

But thinking of a good name is hard. UnaryManyToMany to me feels like I would know better what it all is about. But maybe there may be better names. Or I am just thinking about it wrong.

Also fun times thinking about naming `exp`, two operands, with the first one able to be any size, and the second one exactly one element. :sos:

Reply:
I agree that because of our list semantics, the notion of what is a _single_ operand is a bit complex...

Reply:
Sounds good :) It is also interesting for expressions with more than one operand, where e.g. `mul` can have two lists, but `exp` can have a list as the first operand, but requires a single element as the second.

It would probably be cool to encode that information in the operand types, i.e. `BinaryExpr&ToManyExpr exp(ToManyExpr value, ToOneExpr second)`. It can then allow us to statically check for correctness as well, as per #70 (although there are few cases where it is necessary?). Downside is that we can not pass a list expression where we know that it will only ever contain a single value, so in the `exp` case, it would need a wrap with a `last` for example.